core/app/models/spree/payment/processing.rb

```diff
@@ -94,7 +94,8 @@ def gateway_options
           # payment gateways happy.
           #
           # For more information, please see Spree::Payment#set_unique_identifier
-          order_id: gateway_order_id
+          order_id: gateway_order_id,
+          originator: self
```

Comment:
On further review the complex object is what's expected further down the chain:
https://github.com/solidusio/solidus/blob/v2.0.0/core/app/models/spree/store_credit.rb#L137

I am also not a fan of the generic term originator but that's what's expected here:
https://github.com/solidusio/solidus/blob/v2.0.0/core/app/models/spree/payment_method/store_credit.rb#L36